### PR TITLE
🛠️Execution Refactor & 🎉 Nested Nodes

### PR DESCRIPTION
**Pull request by vanyle**
A lot of new blocks have been added but they are not executable yet !

This pull request aims to:
- Make the new blocks like *slider* executable
- Polish the executing code
- Fix some bugs
- Introduce new node types

While this might seem like multiple features added at once, because they are all deeply intertwined, it makes sense to group them in the same branch.
More specifically, we will try to fix:

- #112
- Tests to prevent #112 regressions
- #94
- Tests to prevent #94 regressions
- #5

### CODE REVIEW

**Comment by vanyle** [comment]
By creating a new kernel when creating a new scene, file loading and file creation is a bit slower.
This could be fix by making the creation of the kernel asynchronous, but this is another issue that is less critical.

So in case somebody in the future wants to increase file opening performance, he should focus on this.

**Comment by vanyle** [comment]
A lot has been done. Nodes can be grouped but nodes inside containers cannot interact with the outside. They don't even share the same kernel ! (And I'd argue this is a good thing as containers might run on different python versions)

**Review by MathisFederico — COMMENTED** [review]
There is some things that we will need to change:

1. Allow to change slider min, max and step, and base scale (linear vs exp?)
image

2. Make nested blocks easier to use.

3. Make output panels always open when raising error, color block when ran / failed.

But those will be for other PRs.
From what has been announced, this PR still lacks of multiple tests.

File: opencodeblocks/blocks/executableblock.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        """Called when the output is an error"""
+        self.has_been_run = False
+
+    @property
+    def source(self) -> str:
+        """Source code"""
+        raise NotImplementedError("source(self) should be overriden")
+
+    @source.setter
+    def source(self, value: str):
```
[the remark this answers is not in the document]
Comment (vanyle) [reply opencodeblocks/blocks/executableblock.py:137]: Didn't want to add another dependency to the project but if you're ok with it, I'm adding it.

[the remark this answers is not in the document]
Comment (vanyle) [reply opencodeblocks/blocks/executableblock.py:137]: Nevermind, abc is built into python by default.

**Comment by vanyle** [comment]
Working on adding the tests right now !

**Review by MathisFederico — APPROVED** [review]
I trust you, I've not had time to review everything as there is too much changes in this PR

**Review by AlexandreSajus — APPROVED** [review]
Looks good to me. run_left and run_right will be modified further in feature/visual-flow

**Comment by vanyle** [comment]
Don't delete the branch ! Nested blocks need to be improved

**Pull request merged**